### ISSUE TITLE
getStepLengths bug if cumulative days appear in second or later cycle

### ISSUE THREAD

**Issue by drewj-usnctech** [terrapower/armi#1415]
Found this in some internal testing where I had a case that had multiple cycles of cumulative day steps. Say I want to deplete with two cycles, each with two 50 day depletion steps.  I can specify this with two ways

Just using `step days`
```python
cs["cycles"] = [
    {"name": "a", "step days": [50, 50]},
    {"name": "b", "step days": [50, 50]},
]
getStepLengths(cs)
[[50.0, 50.0], [50.0, 50.0]]
```
which is correct! Two cycles, each with two 50 day steps ✅ 

Using `cumulative days`
```python
cs["cycles"] = [
    {"name": "a", "cumulative days": [50, 100]},
    {"name": "b", "cumulative days": [150, 200]},
]
getStepLengths(cs)
[[50.0, 50.0], [150.0, 50.0]]
```
which has an incorrect step in the first value of the second cycle

**Comment by john-science** [comment]
Let's see.  The code that generates those lists is here:

https://github.com/terrapower/armi/blob/2dab1222f7ea79d3083db3aabf35b00510bf8d2a/armi/utils/__init__.py#L201-L203

Which calls:

https://github.com/terrapower/armi/blob/2dab1222f7ea79d3083db3aabf35b00510bf8d2a/armi/utils/mathematics.py#L284-L292

So, the confusion here might be what "cumulative" means.

1. In the first case, you define "steps" of days, and then you look at the `getStepLengths()`, and get what you expect.
2. But in the second case, you define the "total number of days that have passed" and then you look at `getStepLengths()`.

You'd expect (1) and (2) to return different results, right?

I have to say, at a glance, this looks correct to me. But what were YOU expecting (2) to return, and why?

**Comment by keckler** [comment]
Yeah, the current interpretation is cumulative within a given cycle. I
think that everything else throughout the code base is consistent with that
interpretation, but please let me know if youve found somewhere that it
isnt.

**Comment by drewj-usnctech** [comment]
Okay, it seems this is because of my misunderstanding / skimming of the docs. `cumulative` does not equal *total cumulative* time, just cumulative within cycle.  

The [cycle history](https://terrapower.github.io/armi/user/inputs/settings.html#cycle-history) example does call this out, but somewhat implicitly

```yaml
power: 1000000
nCycles: 4
cycles:
  - name: A
    step days: [1, 1, 98]
    power fractions: [0.1, 0.2, 1]
    availability factor: 0.1
  - name: B
    cumulative days: [2, 72, 78, 86]
    power fractions: [0.2, 1.0, 0.95, 0.93]
...
```

> Such a scheme would define the following cycles:
> 
> 1. A 2 day power ramp followed by full power operations for 98 days, with three nodes clustered during the ramp and another at the end of the cycle, followed by 900 days of shutdown
> 
>  2. A 2 day power ramp followed by a prolonged period at full power and then a slight power reduction for the last 14 days in the cycle

I say implicitly only because

> the current interpretation is cumulative within a given cycle

is not explicitly stated. It's in the docs, but takes some deciphering.

For some background, my confusion comes from comparing this to e.g., Serpent's [`dep daytot`])(https://serpent.vtt.fi/mediawiki/index.php/Input_syntax_manual#dep_.28depletion_history.29) entry, which is also cumulative. But since there's no concept of "cycles" in  a Serpent entry, I was expecting my example `cycles` input to be similar to something like
```
dep daystep 50 50 50 50
```
or
```
dep daytot 50 100 150 200
```
